diff --git a/Makefile.include b/Makefile.include
index 39fa372..4f95fdd 100644
--- a/Makefile.include
+++ b/Makefile.include
@@ -1,8 +1,22 @@
 BASEPATH_ := $(dir $(realpath $(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST))))
 #BASEPATH_ := $(dir $(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST)))
 
-CC ?= gcc
-CXX ?= g++
+ifneq ($OS(OS), Windows_NT)
+     UNAME_S := $(shell uname -s)
+     ifeq ($(UNAME_S),Darwin)
+		CC ?= /opt/homebrew/opt/gcc/bin/gcc-13
+		CXX ?= /opt/homebrew/opt/gcc/bin/g++-13
+     else
+		CC ?= gcc
+		CXX ?= g++
+     endif
+else # guessing what will work on Windows
+	CC ?= gcc
+	CXX ?= g++
+endif
+
+
+
 SCAN_BUILD ?= scan-build
 PYTHON2 ?= python2
 CPPCHECK ?= cppcheck
